File: 1-Part/3-chapter/palidrome.py
def isPalibrome(theString):
    if len(theString) == 0 or len(theString) == 1:
        return True
    else:
        head = theString[0]
        middle = theString[1: -1]
        last = theString[-1]
        return head == last and isPalibrome(middle)

text = 'racecar'
print(text + 'is a palidrome:' + str(isPalibrome(text)))
text = 'tacocat'
print(text + 'is a palidrome:' + str(isPalibrome(text)))
text = 'xophie'
print(text + 'is a palidrome:' + str(isPalibrome(text)))

File: 1-Part/3-chapter/palidrome.js
function palidrome(theString) {
    if (theString.length === 0 || theString.length === 1){
        return true
    } else {
        let head = theString[0]
        let middle = theString.substring(1, theString.length - 1)
        let last = theString[theString.length - 1]
        return head === last && palidrome(middle)
    }
}

let text = 'racecar'
console.log(`${text} is palidrome: ${palidrome(text)}`)
text = 'tacocat'
console.log(`${text} is palidrome: ${palidrome(text)}`)
text = 'xophie'
console.log(`${text} is palidrome: ${palidrome(text)}`)